src/render_helpers/shaders/border.frag

```diff
@@ -21,6 +23,178 @@ uniform vec2 geo_size;
 uniform vec4 outer_radius;
 uniform float border_width;
 
+float srgb_to_linear(float color) {
+  return 
+    color <= 0.04045 ?
+    color / 12.92 :
+    pow((color + 0.055) / 1.055, 2.4) ;
+}
+
+float linear_to_srgb(float color) {
+  return 
+    color <= 0.0031308 ?
+    color * 12.92 :
+    pow(color * 1.055, 1.0 / 2.4) - 0.055 ;
+}
+
+vec4 lab_to_lch(vec4 color) {
+  float c = sqrt(pow(color.y, 2.0) + pow(color.z, 2.0));
+  float h = degrees(atan(color.z, color.y)) ;
+  h += h <= 0.0 ?
+    360.0 :
+    0.0 ;
+  return vec4(
+    color.x,
+    c,
+    h,
+    color.a
+  );
+}
+
+vec4 lch_to_lab(vec4 color) {
+  float a = color.y * clamp(cos(radians(color.z)), -1.0, 1.0);
+  float b = color.y * clamp(sin(radians(color.z)), -1.0, 1.0);
+  return vec4(
+    color.x,
+    a,
+    b,
+    color.a
+  ); 
+}
+
+vec4 scaling(vec4 color){
+  float divisor =
+    max(max(color.r, color.g), color.b) > 1.0 ?
+    max(max(color.r, color.g), color.b) :
+    1.0 ;
+  color.rgb /= divisor;
+  return color;
+}
+
+vec4 linear_to_oklab(vec4 color){
+  float l = color.r * 0.4122214708 + color.g * 0.5363325363 + color.b * 0.0514459929;
```

Comment:
Idk if it makes any difference in the compiled shader bytecode, but these operations can probably be written as something like:
```
mat3x3 rgb_to_lms = mat3x3(/* the numbers */);
mat3x3 lms_to_lab = mat3x3(/* the numbers */);

vec3 lms = pow(color.rgb * rgb_to_lms, 3.0);
return vec4(lms * lms_to_lab, color.a);
```